server_flask/users.py

```diff
@@ -196,86 +200,76 @@ def get_user_profile():
-        password=couchdb_client_config.password,
+
+    admin_config = CouchDBClientConfig(
+        baseurl=current_app.config["URI_DATABASE"],
+        user=current_app.config["DB_USER"],
+        password=current_app.config["DB_PASSWORD"],
     )
 
-    try:
-        # Open a session as admin
-        admin_session = requests.Session()
-        response = admin_session.post(
-            urljoin(couchdb_client_config.baseurl, "_session"),
-            json={
-                "name": admin_auth.username,
-                "password": admin_auth.password,
-            },
-        )
-        response.raise_for_status()
+    admin_session = _create_session(client_config=admin_config)
 
-        # Ensure the user_name is valid
-        if not _validate_user(user=user_name):
-            return (
-                "This user name is not allowed. Please make sure your user name doesn not start with 'user_' and is less than 32 characters long.",
-                400,
-            )
+    # ID of the user document and its content
+    user_doc_id = "org.couchdb.user:{}".format(requested_user)
 
-        # ID of the user document and its content
-        user_doc_id = "org.couchdb.user:{}".format(user_name)
+    # Check whether the user already exists (e.g., from a previous initialize)
+    response = admin_session.get(
+        urljoin(admin_config.baseurl, "_users/{}".format(user_doc_id)),
+    )
+    response.raise_for_status()
 
-        # Check whether the user already exists (e.g., from a previous initialize)
-        response = admin_session.get(
-            urljoin(couchdb_client_config.baseurl, "_users/{}".format(user_doc_id)),
-        )
-        if response.status_code != 200:
-            # The user does not already exist. Return with 400
-            return "The user does not exist", 400
-        else:
-
-            # The user exists, check if database exists.
-            database = _database_for_user(user=user_name)
-            response = admin_session.head(
-                urljoin(
-                    couchdb_client_config.baseurl, _database_for_user(user=user_name)
-                ),
-            )
-            if response.status_code != 200:
-                # The database does not exist. Something is weird. Raise 500.
-                return (
-                    "Uh oh, this shouldn't happen. Database doesn't exist for this user. Please get in touch with the administrator.",
-                    500,
-                )
-
-            else:
-                # Return the database name since it exists.
-                return {"user_name": user_name, "database": database}
-
-    except requests.exceptions.HTTPError:
-        return (
-            "Uh oh, something went down. Please get in touch with the administrator.",
-            500,
-        )
+    # NOTE: @James: Instead of `response.raise_for_status()`, should we send more usable abort messages if username and/or database doesn't exist? Example provided as comment below.
```

Comment:
@jayfo In `get_user_profile`, instead of `response.raise_for_status()`, should we send more usable/readable abort messages if username and/or database don't exist? Code example provided as comment.